vispy/scene/widgets/grid.py

```diff
@@ -183,63 +258,239 @@ def __repr__(self):
         return (('<Grid at %s:\n' % hex(id(self))) +
                 str(self.layout_array + 1) + '>')
 
-    def _update_child_widgets(self):
-        # Resize all widgets in this grid to share space.
-        n_rows, n_cols = self.grid_size
-        if n_rows == 0 or n_cols == 0:
-            return
+    @staticmethod
+    def _add_total_width_constraints(solver, width_grid, _var_w):
+        for (y, ws) in enumerate(width_grid):
+            width_expr = expression.Expression()
+            for (x, w) in enumerate(ws):
```

Comment:
Yeah, this was with respect to the older code where I was using the `y` and `x` to position.